tests/UnitTests/CoreGraphics.drawing/CGContextDrawingTests.cpp

```diff
@@ -44,6 +44,129 @@ DISABLED_DRAW_TEST_F(CGContext, DrawIntoRect, UIKitMimicTest) {
+    CGContextRef context = GetDrawingContext();
+    woc::unique_cf<CGColorSpaceRef> rgbColorSpace(CGColorSpaceCreateDeviceRGB());
+
+    CGFloat locations[] = { 0.0, 0.5, 1 };
+
+    CGFloat components[] = {
+        1, 0, 0, 1.0, 0, 1, 0, 1.0, 0, 0, 1, 1.0,
+    };
+
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    CGGradientRef gradient = CGGradientCreateWithColorComponents(colorspace, components, locations, 3);
+    CGPoint startPoint = CGPointMake(0, 0);
+    CGPoint endPoint = CGPointMake(512, 1024);
+
+    CGContextDrawLinearGradient(context, gradient, startPoint, endPoint, kCGGradientDrawsBeforeStartLocation);
+}
+
+DRAW_TEST_F(CGContext, LinearGradientWithLowOpacity, UIKitMimicTest) {
+    CGContextRef context = GetDrawingContext();
+    woc::unique_cf<CGColorSpaceRef> rgbColorSpace(CGColorSpaceCreateDeviceRGB());
+
+    CGFloat locations[] = { 0.0, 0.5, 1 };
+
+    CGFloat components[] = {
+        1, 0, 0, 0.1, 0, 1, 0, 0.9, 0, 0, 1, 0.8,
+    };
+
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    CGGradientRef gradient = CGGradientCreateWithColorComponents(colorspace, components, locations, 3);
+    CGPoint startPoint = CGPointMake(300, 750);
+    CGPoint endPoint = CGPointMake(0, 0);
+
+    CGContextDrawLinearGradient(context, gradient, startPoint, endPoint, kCGGradientDrawsBeforeStartLocation);
+}
+
+DRAW_TEST_F(CGContext, LinearGradientWithAlpha, UIKitMimicTest) {
+    CGContextRef context = GetDrawingContext();
+    woc::unique_cf<CGColorSpaceRef> rgbColorSpace(CGColorSpaceCreateDeviceRGB());
+
+    CGFloat locations[] = { 0.0, 0.25, 0.5, 0.6, 0.8, 0.9, 1 };
+
+    CGFloat components[] = {
+        1, 0, 0, 1, 0.4, 0.1, 0.5, 1, 0.50, 0.2, 0.99, 1, 0.41, 0.56, 0, 1, 0.12, 0.12, .3, 1, 0.9, 0.4, 1, 1, 0.2, 0.3, 0.8, 1,
+    };
+
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    CGGradientRef gradient = CGGradientCreateWithColorComponents(colorspace, components, locations, 7);
+    CGPoint startPoint = CGPointMake(512, 1024);
+    CGPoint endPoint = CGPointMake(0, 0);
+
+    CGContextSetAlpha(context, 0.75);
+    CGContextDrawLinearGradient(context, gradient, startPoint, endPoint, kCGGradientDrawsBeforeStartLocation);
+}
+
+DRAW_TEST_F(CGContext, LinearGradientWithLowOpacityShort, UIKitMimicTest) {
+    CGContextRef context = GetDrawingContext();
+    woc::unique_cf<CGColorSpaceRef> rgbColorSpace(CGColorSpaceCreateDeviceRGB());
+
+    CGFloat locations[] = { 0.0, 0.5, 1 };
+
+    CGFloat components[] = {
+        1, 0, 0, 0.8, 0, 1, 0, 0.9, 0, 0, 1, 0.1,
+    };
+
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    CGGradientRef gradient = CGGradientCreateWithColorComponents(colorspace, components, locations, 3);
+    CGPoint startPoint = CGPointMake(250, 300);
+    CGPoint endPoint = CGPointMake(0, 0);
+    CGContextDrawLinearGradient(context, gradient, startPoint, endPoint, kCGGradientDrawsAfterEndLocation);
```

Comment:
going to leave it as, there are too many parameters that will keep on changing and making it will reduce readability.

---
In reply to: [92529089](https://github.com/Microsoft/WinObjC/pull/1535#discussion_r92529089) [](ancestors = 92529089)